New Cask: Puppet Agent

MerelyAPseudonym opened the pull request:
### Changes to a cask
#### Adding a new cask

- [x] Checked there aren’t open [pull requests](https://github.com/caskroom/homebrew-cask/pulls) for the same cask.
- [x] Checked there aren’t closed [issues](https://github.com/caskroom/homebrew-cask/issues) where that cask was already refused.
- [x] When naming the cask, followed the [token reference](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] Commit message includes cask’s name.
- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.

vitorgalvao commented on Casks/puppet-enterprise.rb:
```diff
@@ -0,0 +1,18 @@
+  uninstall launchctl: %w[puppet pxp-agent mcollective],
+            pkgutil:   'com.puppetlabs.puppet-agent'
+
+  zap trash: '~/.puppetlabs'
```
`trash:` doesn’t work, yet. Use `delete:`.

MerelyAPseudonym replied:
Apparently I stopped reading before <q cite="https://github.com/caskroom/homebrew-cask/blob/7f6e06ebfbd91aabed11a99f75c5173e8f882081/doc/cask_language_reference/stanzas/uninstall.md#uninstall-key-trash">It is best not to use this stub until it gains the proper functionality.</q> in the documentation… my bad. I’ll change it.

vitorgalvao commented on Casks/puppet-enterprise.rb:
```diff
@@ -0,0 +1,18 @@
+cask 'puppet-enterprise' do
+  version '2016.1.2'
```
`version '2016.1.2,1.4.2-1'`

MerelyAPseudonym replied:
Interesting. I'll incorporate that.

vitorgalvao commented on Casks/puppet-enterprise.rb:
```diff
@@ -0,0 +1,18 @@
+  version '2016.1.2'
+  sha256 '6af4889773cdb05d638e5d231b87a56612ad092e061cde727d01c965755b4f29'
+
+  url "https://pm.puppetlabs.com/puppet-agent/#{version}/1.4.2/repos/apple/#{MacOS.release}/PC1/x86_64/puppet-agent-1.4.2-1.osx#{MacOS.release}.dmg"
```
url "https://pm.puppetlabs.com/puppet-agent/#{version.before_comma}/version.after_comma.major_minor_patch}/repos/apple/#{MacOS.release}/PC1/x86_64/puppet-agent-#{version.after_comma}.osx#{MacOS.release}.dmg"

vitorgalvao commented on Casks/puppet-enterprise.rb:
```diff
@@ -0,0 +1,18 @@
+
+  conflicts_with cask: %w[puppet facter]
+
+  pkg 'puppet-agent-1.4.2-1-installer.pkg'
```
`pkg "puppet-agent-#{version.after_comma}-installer.pkg"`

vitorgalvao commented:
> Should `~/.puppetlabs` should be removed on zap?

Yes.

> Does the `pkgutil` key to `uninstall` render the `launchctl` key unnecessary? Or should both be left in?

Both should be in.

As for the naming, it is indeed a bit confusing. `puppet-agent` seems best for now, though, as everything on the cask seems to suggest that.

MerelyAPseudonym commented:
@vitorgalvao I’ve come around to thinking that Puppet Agent is indeed the right choice; in fact, I updated the formula to just sidestep “Puppet Enterprise” entirely. However, when inspecting [the earlier build failure](https://travis-ci.org/caskroom/homebrew-cask/jobs/130906319), I realized that the `sha256` I included in the cask would only be correct on El Capitan systems. I updated the formula to include all of the checksums, but something about it is going awry:

```
$ brew cask audit --download puppet-agent
==> Downloading https://downloads.puppetlabs.com/mac/10.11/PC1/x86_64/puppet-age
Already downloaded: /Library/Caches/Homebrew/puppet-agent-1.4.2.dmg
==> Verifying checksum for Cask puppet-agent
audit for puppet-agent: failed
 - a sha256 stanza is required
 - download not possible: sha256 required: sha256 'bb94f785c253614edd2e82e89fdbb4dabe800f0dc009b5658e0d42e11e875ff9'
Error: audit failed
```

vitorgalvao commented:
[Like this](https://github.com/caskroom/homebrew-cask/blob/d41dbf6b83c2e9a2a6d61a5b98d346c11b72604e/Casks/calibre.rb#L2L10).

MerelyAPseudonym commented:
I tried inspecting Travis’s logs, but it’s clear to me what failed.

adidalal commented:
It errors out occasionally, I'll manually pull and check it out.

adidalal commented:
Thank you for the contribution. It needed some fixes, so they were made in 6d2908ceb345d00d7cbfb016f9e053ae5f71820c. Your contribution is still included (and still credited to you), with the appropriate modifications. Please feel free to ask about any of the changes.
